inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/base/ov_layer_test_utils.hpp

```diff
@@ -0,0 +1,123 @@
+
+#include <gtest/gtest.h>
+
+#include <openvino/core/node.hpp>
+#include <openvino/core/function.hpp>
+#include <openvino/pass/manager.hpp>
+#include <openvino/core/type/bfloat16.hpp>
+
+#include <ie_plugin_config.hpp>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "common_test_utils/common_utils.hpp"
+#include "common_test_utils/test_common.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "functional_test_utils/ov_plugin_cache.hpp"
+#include "functional_test_utils/ov_tensor_utils.hpp"
+#include "functional_test_utils/precision_utils.hpp"
+#include "functional_test_utils/layer_test_utils/summary.hpp"
+#include "functional_test_utils/layer_test_utils/environment.hpp"
+
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+#include "ngraph_functions/pass/convert_prc.hpp"
+
+namespace ov {
+namespace test {
+
+using TargetDevice = std::string;
+
+using BasicParams = std::tuple<
+    ov::element::Type,  // Network Precision
+    ov::Shape,          // Input Shape
+    TargetDevice        // Target Device
+>;
+
+enum class RefMode {
+    INTERPRETER,
+    CONSTANT_FOLDING,
+    IE
+};
+
+struct LayerTestsCommon : public CommonTestUtils::TestsCommon {
+    virtual ov::runtime::Tensor generate_input(const ov::element::Type& type, const ov::Shape& shape) const;
+
+    virtual void run();
+
+    virtual void serialize();
+
+    virtual void query_model();
+
+    void compare(const std::vector<std::pair<ov::element::Type, std::vector<std::uint8_t>>>& expected,
+                 const std::vector<ov::runtime::Tensor>& actual);
+
+    void compare(const std::pair<ov::element::Type, std::vector<std::uint8_t>>& expected,
+                 const ov::runtime::Tensor& actual);
+
+    virtual void compare_desc(const ov::runtime::Tensor& expected, const ov::runtime::Tensor& actual);
+
+    virtual void set_ref_mode(RefMode mode);
+
+    std::shared_ptr<ov::Function> get_function();
+
+    std::map<std::string, std::string>& get_configuration();
+
+    std::string get_runtime_precision(const std::string& layerName);
+    std::string get_runtime_precision_by_type(const std::string& layerType);
+
+#ifndef NDEBUG
+    void show_runtime_precision();
+#endif
+
+    LayerTestsCommon();
+
+    virtual void configure_model();
+
+    virtual void compile_model();
+
+    virtual void generate_inputs();
+
+    virtual void infer();
+
+    TargetDevice targetDevice;
+    std::shared_ptr<ngraph::Function> function;
+    std::map<std::string, std::string> configuration;
+    // // Non default values of layouts/precisions will be set to CNNNetwork
+    // InferenceEngine::Layout inLayout = InferenceEngine::Layout::ANY;
+    // InferenceEngine::Layout outLayout = InferenceEngine::Layout::ANY;
+    ov::element::Type inPrc;
+    ov::element::Type outPrc;
+    ov::runtime::ExecutableNetwork executableNetwork;
+    std::vector<ov::runtime::Tensor> inputs;
+    ov::runtime::Core core;
```

Comment:
`ov::runtime::Core core;` -> `std::shared_ptr<ov::runtime::Core> core;`  (I mean use the `ov::test::PluginCache` for it)

Reply:
Why do we need `std::shared_ptr` for Core?

Reply:
why is it nessesry to dereference the cache and create additional copies?